### ISSUE TITLE
Proposal: Email notification - Custom templates

### ISSUE THREAD

**Issue by mind-ar** [authelia/authelia#2756]
### Description
This functionality would allow administrators to modify the contents of the password reset mail
End users may not know it's authelia, and offering a more personalized message could avoid pishing spurs.

the alternatives I can think of are:
1. allow the admin to define a full mail template
2. allow the admin to define some texts (i.e. title, description, etc) 

Let me know what you think about this

### Use Case

_N/A_

**Comment by clems4ever** [comment]
Hello @mind-ar , that's a good idea and it would require a design before implementation. It's not a priority for us to implemenet at the moment though.

**Comment by mind-ar** [comment]
Hi @clems4ever , 
i have two main goals with this feature:
1. avoid too generic password reset mail 
2. add i18n for non english speakers (like me!)

![imagen](https://user-images.githubusercontent.com/10672208/148128823-cea23e85-756f-4499-9ac3-ef65dab58073.png)

Here is some idea of configuration options
```yml
notifier:
  message:
    icon: "/app/icons/acme.png"
    title: "Hello {Firstname}, reset your password"
    body: -|
      This email has been sent to you in order to validate your identity. 
      If you did not initiate the process your credentials might have been compromised. 
      You should reset your password and contact an administrator. 
    footer: "Please contact an administrator if you did not initiate the process. "
   
```
If you agree, i can take this issue 
Let me know what you think.

**Comment by clems4ever** [comment]
Hello @mind-ar,

Regarding, your objectives, I think the first one makes sense but for the second one, it would be better to to document not to use one template per language but instead leverage https://github.com/authelia/authelia/pull/2697 once merged somehow.

On your proposal, I think it would be better to have a dedicated template file for those and then only have the path in the config file. Regarding templating, I recommend using the go template standard feature. And for i18n, I would suggest, the template is written in plain english and then internationalization is applyied on it with your other PR.

WDYT?

**Comment by mind-ar** [comment]
Hi @clems4ever 
Regarding i18n, i totally agree. will see how to implement it after https://github.com/authelia/authelia/pull/2697 is merged

Now, regarding templates,  I'm going to try to describe what I understood
**Method 1 - custom template**
this method follows your suggestion (more or less)
1.  the admin creates a html template like this
```html
<body>
 <h1>{{.title}}</h1>
 hi <b>{{.firstName}}</b>
  <p>This email has been sent to you in order to validate your identity. 
  If you did not initiate the process your credentials might have been compromised. 
  You should reset your password and contact an administrator
  </p>
  <p>the reques was made from: {{.requestIP}} on {{.requestTime}}</p>
  <a href="{{.link}}">click here to continue</a>
</body>
```
configuration section
```yml
notifier:
  template: path_to_template
```

this requires authelia to inject some predefined properties (title, link, ...)


**Method 2 - custom texts in existing template**
This method uses your existing template, and allows the admin to change some sections, as noted in my previous comment

Let me know which method you prefer and, of course, any suggestion/correction.


PD: Apologies, my english is that of a newborn

**Comment by james-d-elliott** [comment]
We discussed internally using a subfolder of the following location such as `email` for the email template overrides. The expected exact file name just needs to be documented, and if it exists it'd just override the appropriate template.
https://github.com/authelia/authelia/blob/master/config.template.yml#L47

Additionally they'd have to provide a full template from the following location (it would probably be acceptable to the team to split the html template into a layout portion and a body portion):

https://github.com/authelia/authelia/tree/master/internal/templates

We also probably want to provide some rudimentary formatting functions for the requestTime.
